Lib/pathlib.py

```diff
@@ -171,7 +171,7 @@ def splitroot(self, part, sep=sep):
                     else:
                         return part[:index2], sep, part[index2+1:]
         drv = root = ''
-        if second == ':' and first in self.drive_letters:
+        if second == ':' and first in self.drive_letters and third == sep:
```

Comment:
The is not the solution. "C:" is a drive-relative path. Of course it has a drive. We need to fix `absolute` to properly support drive-relative paths. See the `FIXME` comment in the code. It's not necessarily as simple as calling `ntpath._getfullpatname`, as suggested in the comment, unless we give up on preserving ".." components in Windows. (That wouldn't be a problem really since preserving ".." components serves no point in Windows, unlike POSIX.)

Reply:
You are right. I will modify the PR.